pkg/services/pluginsintegration/clientmiddleware/caching_metrics.go

```diff
@@ -0,0 +1,35 @@
+package clientmiddleware
+
+import (
+	"github.com/grafana/grafana/pkg/infra/metrics"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	QueryPubdash   = "pubdash"
+	QueryDashboard = "dashboard"
+)
+
+var QueryRequestHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Namespace: metrics.ExporterName,
+	Subsystem: "caching",
+	Name:      "query_request_duration_seconds",
+	Help:      "histogram of grafana query endpoint requests in seconds",
+	Buckets:   []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 25, 50, 100},
+}, []string{"datasource_type", "cache", "query_type"})
+
+var ResourceRequestHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Namespace: metrics.ExporterName,
+	Subsystem: "caching",
+	Name:      "resource_request_duration_seconds",
+	Help:      "histogram of grafana resource endpoint requests in seconds",
+	Buckets:   []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 25, 50, 100},
+}, []string{"datasource_type", "cache"})
```

Comment:
Would suggest naming these metrics to make it clear they're query caching metrics.

Reply:
Sorry missed the subsystem :smile: